### PR TITLE
Make xdgsurfaceitem and xwaylandsurfaceitem public

### PR DESCRIPTION
**Pull request by wineee**

### CODE REVIEW

File: src/server/qtquick/wxdgsurfaceitem.h [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+    void maximumSizeChanged();
+
+private:
+    Q_SLOT void onSurfaceCommit() override;
+    void initSurface() override;
+    QRectF getContentGeometry() const override;
+
+    void setImplicitPosition(const QPointF &newImplicitPosition);
+
+private:
```
Comment (zccrs) [inline src/server/qtquick/wxdgsurfaceitem.h:46]: Move to WXdgSurfaceItemPrivate

File: src/server/qtquick/wxwaylandsurfaceitem.h [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+
+Q_SIGNALS:
+    void parentSurfaceItemChanged();
+    void minimumSizeChanged();
+    void maximumSizeChanged();
+    void positionModeChanged();
+    void positionOffsetChanged();
+    void ignoreConfigureRequestChanged();
+
+private:
```
Comment (zccrs) [inline src/server/qtquick/wxwaylandsurfaceitem.h:65]: The `private` move to WXWaylandSurfaceItemPrivate

**Review by zccrs — CHANGES_REQUESTED** [review]

File: src/server/qtquick/wsurfaceitem.cpp [hunk 3]
```diff
@@ -561,11 +500,17 @@ QWTexture *WSGTextureProvider::ensureTexture()
...
+{
+}
+
+WSurfaceItem::WSurfaceItem(WSurfaceItemPrivate &dd, QQuickItem *parent)
+    : QQuickItem(dd, parent)
 {
     setFlag(ItemIsFocusScope);
 }
 
+
```
Comment (zccrs) [inline src/server/qtquick/wsurfaceitem.cpp]: remove

File: src/server/qtquick/wsurfaceitem.h [hunk 4]
```diff
@@ -108,6 +108,7 @@ class WAYLIB_SERVER_EXPORT WSurfaceItem : public QQuickItem
     Q_ENUM(ZOrder)
 
     explicit WSurfaceItem(QQuickItem *parent = nullptr);
+    explicit WSurfaceItem(WSurfaceItemPrivate &dd, QQuickItem *parent = nullptr);
```
Comment (zccrs) [inline src/server/qtquick/wsurfaceitem.h:111]: move to protected

**Review by zccrs — CHANGES_REQUESTED** [review]

**Review by zccrs — APPROVED** [review]

File: src/server/qtquick/wxwaylandsurfaceitem.h [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+    Q_SLOT void onSurfaceCommit() override;
+    void initSurface() override;
+    bool doResizeSurface(const QSize &newSize) override;
+    QRectF getContentGeometry() const override;
+    QSizeF getContentSize() const override;
+    void geometryChange(const QRectF &newGeometry, const QRectF &oldGeometry) override;
+    Q_SLOT void updatePosition();
```
Comment (wineee) [inline src/server/qtquick/wxwaylandsurfaceitem.h:73]: Should this slot move to WXWaylandSurfaceItemPrivate？

File: src/server/qtquick/wxwaylandsurfaceitem.cpp [hunk 6]
```diff
@@ -0,0 +1,376 @@
...
+        });
+        xwaylandSurface()->safeConnect(&WXWaylandSurface::geometryChanged, this, updateGeometry);
+        connect(this, &WXWaylandSurfaceItem::topPaddingChanged,
+                this, &WXWaylandSurfaceItem::updatePosition, Qt::UniqueConnection);
+        connect(this, &WXWaylandSurfaceItem::leftPaddingChanged,
+                this, &WXWaylandSurfaceItem::updatePosition, Qt::UniqueConnection);
+        // TODO: Maybe we shouldn't think about the effectiveVisible for surface/item's position
+        // This behovior can control by compositor using PositionMode::ManualPosition
+        connect(this, &WXWaylandSurfaceItem::effectiveVisibleChanged,
+                this, &WXWaylandSurfaceItem::updatePosition, Qt::UniqueConnection);
```
Comment (wineee) [inline src/server/qtquick/wxwaylandsurfaceitem.cpp:196]: Not Sure if `UniqueConnection` works for lambda

**Pull request merged**